submit-build-status/action.yml

```diff
@@ -0,0 +1,78 @@
+---
+name: Submit Build Status
+
+description: Submits the build status of a GHA workflow job to CI Analytics. See https://confluence.camunda.com/display/HAN/CI+Analytics
+
+inputs:
+  build_status:
+    description: String representing the build status that should be submitted to CI Analytics, e.g. "success", "failed", "cancelled".
+    required: true
+  user_reason:
+    description: Optional string (200 chars max) the user can submit to indicate the reason why a build has ended with a certain build status , e.g. "flaky-tests".
+    required: false
+    default: ''
+  user_description:
+    description: Optional string (1000 chars max) the user can submit to provide details on the user_reason, e.g. a list of flaky tests.
+    required: false
+    default: ''
+
+  gcp_credentials_json:
+    description: Credentials for a Google Clout ServiceAccount allowed to publish to Big Query formatted as contents of credentials.json file.
+    required: true
+
+runs:
+  using: composite
+  steps:
+  - name: Echo inputs
+    shell: bash
+    run: |
+      echo "Inputs"
+      echo "-----"
+      echo "Build status: ${{ inputs.build_status }}"
+      echo "User reason: ${{ inputs.user_reason }}"
+      echo "User description: ${{ inputs.user_description }}"
+
+  - name: Validate inputs
+    shell: python
+    run: |
+      if '${{ inputs.build_status }}' == '':
+        print('Need to specify a build status that is not empty!')
+        sys.exit(1)
+      if len('${{ inputs.user_reason }}') > 200:
+        print('Specified user reason has more than 200 characters!')
+        sys.exit(1)
+      if len('${{ inputs.user_description }}') > 1000:
+        print('Specified user description has more than 1000 characters!')
+        sys.exit(1)
+
```

Comment:
To access Vault the composite action would need other (additional) inputs explicitly given by the user since it cannot access the {{secrets.}} scope implicitly. That's why I decided against that logic since this way it is more flexible with the user figuring out where to get the secret from - they'd have to do it either way explicitly.

Reply:
just for my own understanding would secret inheritance work in this case? https://docs.github.com/en/actions/using-workflows/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow

In general, we can resolve the conversation

Reply:
Yes, Reusable Workflows support implicit access to `secrets.` (and many other niceties like typed inputs) but can only nest up to N layers of calls (N=4 currently afair) so that could potentially limit teams.